main.tf

```diff
@@ -4,6 +4,14 @@
 # Creates Secret Group within existing Secret Manager instance
 ##############################################################################
 
+locals {
+  # Validation (approach based on https://github.com/hashicorp/terraform/issues/25609#issuecomment-1057614400)
+  # tflint-ignore: terraform_unused_declarations
+  validate_create_access_group = (var.create_access_group && var.access_group_roles == null) ? tobool("When creating an access group, a list of roles must be set") : true
```

Comment:
Can we bump the required tf version to `>=1.9.0` and use cross variable validation in the variables.tf